### PR TITLE
presence-nvim: init plugin

### PR DESCRIPTION
**Pull request by SaiProton**
Adds plugin for Discord Rich Presence for Neovim: 
https://github.com/andweeb/presence.nvim

Also adds tests for this plugin.

### CODE REVIEW

File: plugins/utils/presence-nvim.nix [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+          Update activity based on autocmd events.
+          If `false`, map or manually execute
+          `:lua package.loaded.presence:update()`
+        '';
+
+        neovimImageText = helpers.defaultNullOpts.mkStr ''"The One True Text Editor"'' ''
+          Text displayed when hovered over the Neovim image.
+        '';
+
+        mainImage = helpers.defaultNullOpts.mkStr ''"neovim"'' ''
```
Comment (GaetanLepage) [inline plugins/utils/presence-nvim.nix:41]:
```suggestion
        mainImage = helpers.defaultNullOpts.mkStr "neovim" ''
```
same here

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Hey !
Thank you for contributing :)

First, here are a minor modifications.

**Comment by SaiProton** [comment]
Thanks for the tips, applied the changes :)

**Comment by GaetanLepage** [comment]
Ok, thanks for addressing my comments.
I would suggest to remove your `staticOrFunc` implementation and do something simpler:

```nix
editingText = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.rawType) ''"Editing %s"'' ''
  String rendered when `enableLineNumber` is set to `true` to display the current line number.
  Can also be a  lua function:
  `function(line_number: number, line_count: number): string`.
''
```

**Comment by GaetanLepage** [comment]
In several other plugins, we rely on the user to use the `foo.__raw = "lua code"` syntax.

**Review by GaetanLepage — APPROVED** [review]
Great ! Now it is good for me :)

**Comment by SaiProton** [comment]
Yeah, I didn't actually know you could actually specify raw types like that which is cool.

Also, when I updated the `buttons` option to be able to accept either a raw type or list of submodules, I noticed the documentation for the submodule attributes no longer appeared in the generated docs, so I specified the docs for those attributes under `buttons` if that's okay.

**Pull request merged**